persistence/src/main/java/com/anytypeio/anytype/persistence/repo/DefaultUserSettingsCache.kt

```diff
@@ -300,6 +300,54 @@ class DefaultUserSettingsCache(
         }
     }
 
+    override suspend fun setLastSearchQuery(query: String, space: SpaceId) {
+        context.spacePrefsStore.updateData { existingPreferences ->
+            val givenSpacePreference = existingPreferences
+                .preferences
+                .getOrDefault(
+                    key = space.id,
+                    defaultValue = SpacePreference()
+                )
+            val updated = givenSpacePreference.copy(
+                lastSearchQuery = query
+            )
+            val result = buildMap {
+                putAll(existingPreferences.preferences)
+                put(key = space.id, updated)
+            }
+            SpacePreferences(preferences = result)
+        }
+    }
+
+    override suspend fun getLastSearchQuery(space: SpaceId): String {
+        return context.spacePrefsStore
+            .data
+            .map { preferences ->
+                preferences
+                    .preferences[space.id]
+                    ?.lastSearchQuery.orEmpty()
+            }
+            .first()
```

Comment:
`firstOrNull` might be safier

Reply:
the error is handled everywhere, so this is the expected behavior